src/main/java/net/imagej/legacy/plugin/MacroAutoCompletionProvider.java

```diff
@@ -335,24 +342,42 @@ private void appendMacroSpecificCompletions(String input, List<Completion> resul
 				String command = trimmedline.substring(8).trim().replace("{", "");
 				String lcasecommand = command.toLowerCase();
 				if (lcasecommand.contains(lcaseinput)) {
-					String description = findDescription(textArray, linecount, "User defined function " + command + "\n as specified in line " + (linecount + 1));
+					String description = findDescription(textArray, linecount, "User defined function <b>" + command + "</b> as specified in line " + (linecount + 1));
 
 
 					completions.add(new BasicCompletion(this, command, null, description));
 				}
 			}
-			if (lcaseline.contains("=")) {
+			if ((lcaseline.contains("=")) && (charcount < codeLength)) {
 				String command = trimmedline.substring(0, lcaseline.indexOf("=")).trim();
+				boolean globalVar = false;
+				if(command.startsWith("var ")) { 
+					command = command.substring(4).trim();
+					globalVar= true;
+				}
 				String lcasecommand = command.toLowerCase();
-				if (lcasecommand.contains(lcaseinput) && command.matches("[_a-zA-Z]+")) {
-					String description = "User defined variable " + command + "\n as specified in line " + (linecount + 1);
-
-					completions.add(new BasicCompletion(this, command, null, description));
+				if (lcasecommand.contains(lcaseinput) && command.matches("([_a-zA-Z]+[_a-zA-Z0-9]*|[_a-zA-Z0-9]*[a-zA-Z]+[a-zA-Z0-9])")) { // [_a-zA-Z0-9]+,  excluding [0-9]+[_0-9]*
```

Comment:
Would you mind leaving a comment here telling what the regexp stands for? I'm one of the people who cannot read regexp ;-)

Reply:
Sorry for the cryptic regex (and regex as comment!). In fact, the regex in the comment is far more easily read than the code itself. Besides, I also realized that one of the cases I was ensuring is not really valid (e.g. "3stooges"), so I can be much simplified: 

```suggestion
				if (lcasecommand.contains(lcaseinput) && command.matches("[_a-zA-Z]+[_a-zA-Z0-9]*")) { // First character cannot be a digit ([_a-zA-Z]+), while the rest can be any valid character ([_a-zA-Z0-9]*)
```